### DISCUSSION TITLE
Feature Request: add useDisclosure Hook , to makes controlling dialogs easier

### DISCUSSION THREAD

**Discussion by deaa1111** [primefaces/community#1927] [PrimeReact]
```
const function useDisclose(initState?: boolean) {
    const [isOpen, setIsOpen] = React.useState(initState || false);
    const onOpen = () => {
        setIsOpen(true);
    };
    const onClose = () => {
        setIsOpen(false);
    };
    const onToggle = () => {
        setIsOpen(!isOpen);
    };
    return {
        isOpen,
        onOpen,
        onClose,
        onToggle
    };
}
```

**Comment by nitrogenous** [comment]
Thank you for your interest in the PrimeReact library. We appreciate your innovative suggestion. At present, our team is focused on upgrading the library to its next-generation version. Consequently, we have temporarily paused the addition of new features and components. However, please be assured that your idea has been noted and will be considered for inclusion in the forthcoming next-generation PrimeReact library.
